### ISSUE TITLE
color-contrast integration tests - Shadow DOM support

### ISSUE THREAD

**Issue by marcysutton** [dequelabs/axe-core#394]
Failing integration tests for the [color contrast rule integration](https://github.com/dequelabs/axe-core/blob/develop/test/integration/rules/color-contrast/color-contrast.html):
```
color-contrast
    color-contrast test
      passes
        1) should find ["#pass1"]
        2) should find ["#pass2"]
        3) should find ["#pass3"]
        4) should find ["#pass4"]
        5) should find ["#pass5"]
        6) should find ["#pass6"]
        7) should find ["#pass6 > input[type=\"text\"]"]
        8) should find ["#pass7"]
        9) should find ["#pass7 > input"]
        10) should not return other results
      violations
        11) should find ["#fail1"]
        12) should find ["#fail2"]
        13) should find ["#fail3"]
        14) should find ["#fail6"]
        15) should not return other results
      incomplete
        16) should find ["#canttell1"]
        17) should find ["#canttell2"]
        18) should find ["#canttell3"]
        19) should find ["#canttell4"]
        20) should find ["#canttell5"]
        21) should find ["#canttell6"]
        22) should find ["#canttell7"]
        23) should find ["#canttell8"]
        24) should find ["#canttell9"]
        25) should find ["#canttell10"]
        26) should find ["#canttell11"]
        27) should find ["#canttell12"]
        28) should find ["#canttell13"]
        29) should find ["#canttell14"]
        30) should find ["#canttell15"]
        31) should find ["#canttell16"]
        32) should find ["#canttell17"]
        33) should find ["#canttell18"]
        34) should find ["#canttell19"]
```

Subtask of #87. 

Full test result:
```
1) color-contrast color-contrast test passes should find ["#pass1"]:
     there are no passes
  

  2) color-contrast color-contrast test passes should find ["#pass2"]:
     there are no passes
  

  3) color-contrast color-contrast test passes should find ["#pass3"]:
     there are no passes
  

  4) color-contrast color-contrast test passes should find ["#pass4"]:
     there are no passes
  

  5) color-contrast color-contrast test passes should find ["#pass5"]:
     there are no passes
  

  6) color-contrast color-contrast test passes should find ["#pass6"]:
     there are no passes
  

  7) color-contrast color-contrast test passes should find ["#pass6 > input[type=\"text\"]"]:
     there are no passes
  

  8) color-contrast color-contrast test passes should find ["#pass7"]:
     there are no passes
  

  9) color-contrast color-contrast test passes should find ["#pass7 > input"]:
     there are no passes
  

  10) color-contrast color-contrast test passes should not return other results:
     there are no passes
  

  11) color-contrast color-contrast test violations should find ["#fail1"]:
     there are no violations
  

  12) color-contrast color-contrast test violations should find ["#fail2"]:
     there are no violations
  

  13) color-contrast color-contrast test violations should find ["#fail3"]:
     there are no violations
  

  14) color-contrast color-contrast test violations should find ["#fail6"]:
     there are no violations
  

  15) color-contrast color-contrast test violations should not return other results:
     there are no violations
  

  16) color-contrast color-contrast test incomplete should find ["#canttell1"]:
     Element not found
  

  17) color-contrast color-contrast test incomplete should find ["#canttell2"]:
     Element not found
  

  18) color-contrast color-contrast test incomplete should find ["#canttell3"]:
     Element not found
  

  19) color-contrast color-contrast test incomplete should find ["#canttell4"]:
     Element not found
  

  20) color-contrast color-contrast test incomplete should find ["#canttell5"]:
     Element not found
  

  21) color-contrast color-contrast test incomplete should find ["#canttell6"]:
     Element not found
  

  22) color-contrast color-contrast test incomplete should find ["#canttell7"]:
     Element not found
  

  23) color-contrast color-contrast test incomplete should find ["#canttell8"]:
     Element not found
  

  24) color-contrast color-contrast test incomplete should find ["#canttell9"]:
     Element not found
  

  25) color-contrast color-contrast test incomplete should find ["#canttell10"]:
     Element not found
  

  26) color-contrast color-contrast test incomplete should find ["#canttell11"]:
     Element not found
  

  27) color-contrast color-contrast test incomplete should find ["#canttell12"]:
     Element not found
  

  28) color-contrast color-contrast test incomplete should find ["#canttell13"]:
     Element not found
  

  29) color-contrast color-contrast test incomplete should find ["#canttell14"]:
     Element not found
  

  30) color-contrast color-contrast test incomplete should find ["#canttell15"]:
     Element not found
  

  31) color-contrast color-contrast test incomplete should find ["#canttell16"]:
     Element not found
  

  32) color-contrast color-contrast test incomplete should find ["#canttell17"]:
     Element not found
  

  33) color-contrast color-contrast test incomplete should find ["#canttell18"]:
     Element not found
  

  34) color-contrast color-contrast test incomplete should find ["#canttell19"]:
     Element not found
```

**Comment by marcysutton** [comment]
Closed with #416.
